libs/react-helpers/src/hooks/use-resize-observer.ts

```diff
@@ -0,0 +1,43 @@
+import { captureException } from '@sentry/react';
+import debounce from 'lodash/debounce';
+import { useEffect, useState } from 'react';
+
+type ResizeObserverConfiguration = {
+  debounceTime: number;
+  config: ResizeObserverOptions;
+};
+
+const DEFAULT_OPTIONS: ResizeObserverConfiguration = {
+  debounceTime: 0,
+  config: {
+    box: 'border-box',
+  },
+};
+
+export function useResizeObserver(
+  target: Element | null,
+  callback: ResizeObserverCallback,
+  options: ResizeObserverConfiguration = DEFAULT_OPTIONS
+) {
+  const [observer, setObserver] = useState<ResizeObserver | null>(null);
+
+  useEffect(() => {
```

Comment:
suggestion (non blocking): same as with useMutationObserver, this could be combined into a single effect